src/Console/AllInOneCommand.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace SilverStripe\Upgrader\Console;
+
+use SilverStripe\Upgrader\Util\CommandRunner;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Command to try to update a composer file to use SS4.
+ */
+class AllInOneCommand extends AbstractCommand
+{
+    use FileCommandTrait;
+
+    protected function configure()
+    {
+        $this->setName('all-in-one')
+            ->setDescription('Aggregate all the commands required to upgrade a SilverStripe project.')
+            ->setDefinition([
+                $this->getRootInputOption(),
+                new InputOption(
+                    'strict',
+                    'S',
+                    InputOption::VALUE_NONE,
+                    'Prefer ~ to ^ avoid accidental updates'
+                ),
+                new InputOption(
+                    'recipe-core-constraint',
+                    'R',
+                    InputOption::VALUE_OPTIONAL,
+                    'Version of `silverstripe/recipe-core` you are targeting. Defaults to the last stable',
+                    '*'
+                ),
+                new InputOption(
+                    'composer-path',
+                    'P',
+                    InputOption::VALUE_OPTIONAL,
+                    'Path to your composer executable.',
+                    ''
+                ),
+                new InputOption(
+                    'skip-namespace',
+                    null,
+                    InputOption::VALUE_NONE,
+                    'Skip the `add-namespace` command.'
+                ),
+                new InputOption(
+                    'namespace',
+                    'N',
+                    InputOption::VALUE_OPTIONAL,
+                    'Path to your composer executable.',
+                    'App\\Web'
```

Comment:
I'm not sure we should apply a default namespace here; It's better if it skips this step if not provided.

Reply:
I removed the default namespace. I made the `--namespace` flag mandatory unless you have the `--skip-add-namepsace`. The idea is to nudge people into using namespace.

I've also added the ability to use the --psr4 flag.